runtime/compiler/aarch64/runtime/PicBuilder.spp

```diff
@@ -132,6 +132,22 @@
 	.text
 	.align 2
 
+
+#if defined(OMR_GC_COMPRESSED_POINTERS)
+#if defined(OMR_GC_FULL_POINTERS)
+#define LOAD_CLASS(dest64, dest32, src) \
+	ldr	dest64, [x19, J9TR_VMThreadCompressObjectReferences] ; \
+	cmp	dest64, 0 ; \
+	ldr	dest64, [src, J9TR_J9Object_class] ; \
+	beq	8 ; \
+	ldr	dest32, [src, J9TR_J9Object_class]
```

Comment:
If we expect the 32-bit (compressed) case to be the more common one (which we do), why don't we flip this branch around so that only a single 32-bit load will happen in the common case.  The way it is implemented will always do two loads for the most common case.

Reply:
I agree. I think the common path would be the compressed one.